config.js

```diff
@@ -228,216 +217,159 @@ exports.updateFromArgv = function() {
         .describe('minperiodtimecorrunranked', 'Minimum seconds of period time for correspondence unranked games')
         .describe('maxperiodtimecorrunranked', 'Maximum seconds of period time for correspondence unranked games')
     ;
-    const argv = optimist.argv;
 
-    if (!argv._ || argv._.length === 0) {
+    if (!optimist.argv._ || optimist.argv._.length === 0) {
         optimist.showHelp();
         process.exit();
     }
 
     // console messages
     // A- greeting and debug status //
     let debugStatusMessage = "OFF";
-    if (argv.debug) {
+    if (optimist.argv.debug) {
         debugStatusMessage = "ON\n-Will show detailed debug booting data\n-Will show all console notifications";
     }
     console.log(`\ngtp2ogs version 6.0\n--------------------\n- For changelog or latest devel updates, please visit https://github.com/online-go/gtp2ogs/tree/devel\nDebug status: ${debugStatusMessage}`);
     // B - check deprecated argv //
     testDeprecatedArgv(optimist.argv, "komis");
 
+    /* exports arrays
+    /  A) general case:*/
+    const genericMain_r_u_Families = ["minmaintimeblitz","minmaintimelive",
+    "minmaintimecorr","maxmaintimeblitz", "maxmaintimelive", "maxmaintimecorr", "minperiodsblitz",
+    "minperiodslive", "minperiodscorr", "maxperiodsblitz", "maxperiodslive", "maxperiodscorr",
+    "minperiodtimeblitz", "minperiodtimelive", "minperiodtimecorr", "maxperiodtimeblitz",
+    "maxperiodtimelive", "maxperiodtimecorr", "minhandicap", "maxhandicap", "noautohandicap",
+    "proonly", "nopauseonweekends", "nopause"];
+
+    /* B) specific cases:
+    /  rank family args need parsing before exporting*/
+    const rank_r_u_Families = ["minrank", "maxrank"];
+    // bans family is an example of All/ranked/unranked family:
+    // export general AND ranked AND unranked args
+    const all_r_u_Families = ["bans"];
+    // note: allowed_r_u_Families use a different formula before
+    //       exporting, included here as well
+
+    // C) combinations of all r_u args, to export separately
+    const full_r_u_Families = genericMain_r_u_Families
+                              .concat(rank_r_u_Families)
+                              .concat(all_r_u_Families)
+                              .concat(allowed_r_u_Families);
```

Comment:
Actually you can use concat once and give it multiple arrays.

```
    const full_r_u_Families = genericMain_r_u_Families.concat(
                              rank_r_u_Families,
                              all_r_u_Families,
                              allowed_r_u_Families);
```

Reply:
done, nice indeed :100: